packages/client3/components/pool/pool-information/pool-information.tsx

```diff
@@ -0,0 +1,333 @@
+import { Progress } from "@/components/design-system/progress";
+import { WalletButton } from "@/components/design-system/wallet-button";
+import { TRANCHES, USDC_DECIMALS } from "@/constants";
+import { useUser } from "@/hooks/user-hooks";
+import { generateErc20PermitSignature, useContract } from "@/lib/contracts";
+import {
+  SupplyPanelUserFieldsFragment,
+  UidType,
+} from "@/lib/graphql/generated";
+import {
+  approveErc20IfRequired,
+  signAgreement,
+  usdcWithinEpsilon,
+} from "@/lib/pools";
+import { openVerificationModal } from "@/lib/state/actions";
+import { toastTransaction } from "@/lib/toast";
+import { isSmartContract, useWallet } from "@/lib/wallet";
+
+type Props = {
+  totalSuppliedAmount: number;
+  totalGoalAmount: number;
+  totalBackers: number;
+  totalEarned: number;
+  closingDate: Date;
+  tranchedPoolAddress: string;
+  allowedUidTypes: UidType[];
+  remainingJuniorCapacity: BigNumber;
+  type?: "completed" | "failed";
+  agreement?: string;
+};
+
+interface FormFields {
+  supply: string;
+  backerName: string;
+}
+
+export function PoolInformation({
+  totalSuppliedAmount,
+  totalGoalAmount,
+  totalBackers,
+  totalEarned,
+  closingDate,
+  type,
+  agreement,
+  tranchedPoolAddress,
+  allowedUidTypes,
+  remainingJuniorCapacity,
+}: Props) {
+  const apolloClient = useApolloClient();
+  const { account, provider } = useWallet();
+  const tranchedPoolContract = useContract("TranchedPool", tranchedPoolAddress);
+  const usdcContract = useContract("USDC");
+  const user = useUser() as SupplyPanelUserFieldsFragment;
+
+  // const isUserVerified =
+  //   user?.isGoListed ||
+  //   user?.isUsEntity ||
+  //   user?.isNonUsEntity ||
+  //   user?.isUsAccreditedIndividual ||
+  //   user?.isUsNonAccreditedIndividual ||
+  //   user?.isNonUsIndividual;
+
+  // const canUserParticipate = user
+  //   ? canUserParticipateInPool(allowedUidTypes, user)
+  //   : false;
+  /** TODO: REMOVE -> JUST TO TEST */
+  const isUserVerified = true;
+  const canUserParticipate = true;
+
+  /** TODO: REMOVE <- JUST TO TEST */
+
+  const rhfMethods = useForm<FormFields>({
+    mode: "onSubmit",
+    shouldFocusError: true,
+  });
+  // const { control, register, formState } = rhfMethods;
+  const { control, watch, register, setValue } = rhfMethods;
+  const [availableBalance, setAvailableBalance] = useState(BigNumber.from(0));
+
+  const validateMaximumAmount = async (value: string) => {
+    if (!account || !usdcContract) {
+      return;
+    }
+    const valueAsUsdc = utils.parseUnits(value, USDC_DECIMALS);
+    console.log(valueAsUsdc);
+    if (valueAsUsdc.gt(remainingJuniorCapacity)) {
+      return "Amount exceeds remaining junior capacity";
+    }
+    if (valueAsUsdc.lte(BigNumber.from(0))) {
+      return "Must deposit more than 0";
+    }
+    if (
+      valueAsUsdc.gt(availableBalance) &&
+      !usdcWithinEpsilon(valueAsUsdc, availableBalance)
+    ) {
+      return "Amount exceeds USDC balance";
+    }
```

Comment:
These validations could be extracted out for reusability I guess?

Reply:
fixed